controller/src/test/java/io/pravega/controller/server/v1/InMemoryControllerServiceImplTest.java

```diff
@@ -85,9 +86,10 @@ public void setup() throws Exception {
 
         Cluster mockCluster = mock(Cluster.class);
         when(mockCluster.getClusterMembers()).thenReturn(Collections.singleton(new Host("localhost", 9090, null)));
-        controllerService = new ControllerServiceImpl(
```

Comment:
Why can't you *spy* directly this *controllerService* variable?

Reply:
controllerService which is an instance of `ControllerServiceImpl` is what we are verifying in this test.

Reply:
I am not sure this is the right way. But unfortunately I don't have much time to suggest another approach. 
So I am fine with keeping it.

Maybe one day we could introduce PowerMock, that allows you to rewrite entire methods and constructors without introducing this intermediate objects.

Now we are introducing this layer between the streamStore and the ControllerService, I don't have enough context, so I don't know if it may introduce some sort of difference from the behavior of the tests  and the execution at runtime in production (the risk is to not test the right thing).